interfaces/apparmor/backend.go

```diff
@@ -198,6 +198,13 @@ func snapConfineFromCoreProfile(coreInfo *snap.Info) (dir, glob string, content
 		vanillaProfileText, []byte("/usr/lib/snapd/snap-confine"), []byte(snapConfineInCore), -1)
 	// /snap/core/111/usr/lib/snapd/snap-confine -> snap.core.111.usr.lib.snapd.snap-confine
 	patchedProfileName := strings.Replace(snapConfineInCore[1:], "/", ".", -1)
+	// we need to add a uniqe prefix on core18, otherwise we generate
+	// something like snap.snapd.56.usr.lib.snapd.snap-confine which
+	// will collide with the regular snapd snap profile glob
+	if !release.OnClassic {
+		patchedProfileName = "special." + patchedProfileName
```

Comment:
As discussed on IRC we can install our existing `snapd.apparmor.service` on core and then patch it to consider a new directory `/var/lib/snapd/apparmor/internal` (path is TBD) so that we are both not clashing with any snap and also not constrained by apparmor init scripts.

Reply:
As discussed on irc I will go with a new prefix `snap-confine.*` (or alternatively `system.*`).